syntax = "proto3";

package rglc;

message KeepAliveRequest {}
message KeepAliveResponse {}

message BuildGeoListsRequest {
    bool is_use_whitelist = 1;
    repeated string formats = 2;
    repeated string presets = 3;
    string out_dir = 4;
}

message BuildGeoListsResponse {
    repeated string geoip_paths = 1;
    repeated string geosite_paths = 2;
    string release_notes_path = 3;
    bool is_empty = 4;
}

service RGLC_Service {
  rpc SendKeepAlive(KeepAliveRequest) returns (KeepAliveResponse);
  rpc BuildGeoLists(BuildGeoListsRequest) returns (BuildGeoListsResponse);
}
